modules/services/batsignal.nix

```
@@ -0,0 +1,181 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.batsignal;
+
+  commandLine = concatStringsSep " " ([ "${cfg.package}/bin/batsignal" ]
+    ++ optional (cfg.warningLevelPercent != null)
+    "-w ${toString cfg.warningLevelPercent}"
+    ++ optional (cfg.criticalLevelPercent != null)
+    "-c ${toString cfg.criticalLevelPercent}"
+    ++ optional (cfg.criticalLevelPercent != null)
+    "-d ${toString cfg.dangerLevelPercent}"
+    ++ optional (cfg.fullLevelPercent != null)
+    "-f ${toString cfg.fullLevelPercent}"
+    ++ optional (cfg.warningLevelMessage != null)
+    "-W '${cfg.warningLevelMessage}'"
+    ++ optional (cfg.criticalLevelMessage != null)
+    "-C '${cfg.criticalLevelMessage}'"
+    ++ optional (cfg.dangerLevelCommand != null) (let
+      cmd = pkgs.writeShellScript "batsignal-dangercmd" cfg.dangerLevelCommand;
+    in "-D ${cmd}")
+    ++ optional (cfg.fullLevelMessage != null) "-F '${cfg.fullLevelMessage}'"
+    ++ optional (cfg.batteryNames != null)
+    "-n '${concatStringsSep "," cfg.batteryNames}'"
+    ++ optional (cfg.updateIntervalSeconds != null)
+    "-m ${toString cfg.updateIntervalSeconds}"
+    ++ optional (cfg.appName != null) "-a '${cfg.appName}'"
+    ++ optional (cfg.icon != null) "-I ${cfg.icon}");
+
+in {
+  meta.maintainers = with maintainers; [ loicreynier ];
+
+  options.services.batsignal = {
+    enable = mkEnableOption "batsignal battery monitor daemon";
+    package = mkPackageOption pkgs "batsignal" { };
+
+    warningLevelPercent = mkOption {
+      type = types.nullOr (types.ints.between 0 100);
+      default = null;
+      example = 15;
```

I think we should line up these defaults with what batsignal does: https://github.com/electrickite/batsignal/blob/master/batsignal.1.in#L34

```suggestion
      default = 15;
      example = 20;
```